bittensor/_subtensor/subtensor_impl.py

```diff
@@ -1500,6 +1513,91 @@ def neurons(self, block: int = None ) -> List[SimpleNamespace]:
                 break
         return neurons
 
+    def neurons_fast(self, block: int = None ) -> List[SimpleNamespace]: 
+        r""" Returns a list of neuron from the chain, using the bundled subtensor-node-api
+        Args:
+            block (int):
+                block to sync from.
+        Returns:
+            neuron (List[SimpleNamespace]):
+                List of neuron objects.
+        """
+        neurons = []
+        if block is None:
+            block: int = self.get_current_block()
+
+        block_hash: str = self.substrate.get_block_hash( block )
+        endpoint_url: str = self.chain_endpoint
+        # format endpoint with ws://
+        if "ws://" not in endpoint_url and "wss://" not in endpoint_url:
+            endpoint_url = "ws://" + endpoint_url
+        
+        try:
+            if platform == "linux" or platform == "linux2":
+                # linux
+                os_name = "linux"
+            elif platform == "darwin":
+                # OS X
+                os_name = 'macos'
+            else:
+                raise Exception('Unsupported platform for subtensor-node-api')
+            bittensor.__console__.print("Using subtensor-node-api for neuron retrieval...")
+            path_to_bin = os.path.join(os.path.dirname(__file__), f"../../bin/subtensor-node-api-{os_name}")
```

Comment:
Since this is something independent and has importance here I would say that it would be a good idea to extract it to an independent function and test it with unit tests